cpg-core/src/main/kotlin/de/fraunhofer/aisec/cpg/passes/SymbolResolver.kt

```diff
@@ -479,6 +516,77 @@ open class SymbolResolver(ctx: TranslationContext) : ComponentPass(ctx) {
             return
         }
 
+        // Let's do some next generation stuff here, but currently with some restrictions
```

Comment:
I would prefere a comment detaching this from the next gen such that it is more understandable in the future and renaming the variable to useNonLegacyResolution or negating the bools and call it `useLegacyResolution`